webapp/components/at_mention/at_mention.jsx

```diff
@@ -48,32 +72,54 @@ export default class AtMention extends React.PureComponent {
         return '';
     }
 
-    search = (e) => {
-        e.preventDefault();
+    getProfileByUsername(username) {
+        if (!username) {
+            return '';
+        }
+
+        let profile = UserStore.getProfileByUsername(username);
+        if (!profile) {
+            this.props.actions.getUserByUsername(username)(dispatch, getState).then(
+                (data) => {
+                    if (data) {
+                        profile = data;
+                    }
+                }
+            );
+        }
 
-        this.props.actions.searchForTerm(this.state.username);
+        return profile;
     }
 
     render() {
         const username = this.state.username;
+        const user = this.state.user;
 
-        if (!username) {
+        if (!username || !user) {
             return <span>{'@' + this.props.mentionName}</span>;
         }
 
         const suffix = this.props.mentionName.substring(username.length);
 
         return (
-            <span>
-                <a
-                    className='mention-link'
-                    href='#'
-                    onClick={this.search}
-                >
-                    {'@' + username}
-                </a>
-                {suffix}
-            </span>
+            <OverlayTrigger
+                ref='overlay'
+                trigger='click'
+                placement='right'
+                rootClose={true}
+                overlay={
+                    <ProfilePopover
+                        user={user}
+                        src={Client4.getProfilePictureUrl(user.id, user.last_picture_update)}
+                        hide={this.hideProfilePopover}
+                    />
+                }
+            >
+                <span>
+                    <a className='mention-link'>{'@' + username}</a>
+                    {suffix}
```

Comment:
The suffix shouldn't be part of the `OverlayTrigger` since it's not part of the username.